server.py

```diff
@@ -46,8 +44,13 @@ def login():
     # Generate and store a nonce to uniquely identify this login request.
     # This allows us to prevent identity tokens from being used more than once.
     nonce = uuid4().hex
-    expiry = datetime.utcnow() + timedelta(minutes=30)
-    NONCES[nonce] = expiry.timestamp()
+
+    # Wrap Redis SET/EXPIRE in a MULTI/EXEC transaction to ensure both happen.
+    # Without the transaction, we risk adding a nonce but not its expiration.
+    txn = REDIS.pipeline()
+    txn.set(nonce, '')
+    txn.expire(nonce, 15 * 60)  # 15 minutes
+    txn.execute()
```

Comment:
The expiry can be passed as an optional param to `set()`: https://github.com/jamesls/fakeredis/blob/4bfc7a2793aa93723c79512f8be4df2c497498ef/fakeredis.py#L433

Which would also make the transaction unnecessary.